### DISCUSSION TITLE
Update on table with composite primary key slow on MariaDb

### DISCUSSION THREAD

**Discussion by stefaanMLB** [mikro-orm/mikro-orm#4670] [Q&A]
Because of how the queries are constructed, I get bad performance on updating an entity with composite primary key on MariaDb database
The primary key for our `patient` table is combined of the `dossier` and `doc1` fields.
On update, Mikro-orm creates this query
```sql
update `rsa`.`patient` set `birth` = '20001102' where (`dossier`, `doc1`) in (('10279478', 'IX4')) [took 2073 ms]
```
which takes 2 seconds to execute while the below query only takes a few milliseconds to complete
```sql
update `rsa`.`patient` set `birth` = '20001102' where `dossier` = '10279478' and `doc1`= 'IX4';
```
Is it possible to instruct Mikro-orm to build the second type of query, even when a lot of the patient properties change ?

**Comment by B4nan** [comment]
Hmm, and you have a primary key over those two columns (on schema level)? In the correct order?

There is no toggle to control this, but it feels weird that it would be so slow if you have the right indexes. Sounds a bit like a bug in mariadb, not sure if mysql behaves the same. How large is your table?

**Comment by stefaanMLB** [comment]
There's 1605k row in that table.
The same difference when executing the queries in the CLI.
I'll ask the question to the MariaDB team.
In meantime, I solved it by using the QueryBuilder.
